website/content/docs/secrets/terraform.mdx

```diff
@@ -152,18 +152,78 @@ token           <example token>
 token_id        at-fqvtdTQ5kQWcjUfG
 ```
 
-Please see the [HCP Terraform API
-Token documentation for more
-information](/terraform/cloud-docs/users-teams-organizations/api-tokens).
+### Team roles
 
-## Tutorial
+The HCP Terraform secret engine creates dynamic Team API tokens by 
+configuring a Vault role to manage an existing HCP Terraform Team. 
+The lifecycle of these tokens is managed by Vault and/or HCP Terraform.
+Generally, Vault aims to manage tokens in external APIs with revoke actions
+taken according to ttl & max_ttl settings. In addition to that behavior,
+using the max_ttl Vault will set an `ExpiredAt` date in HCP Terraform 
+also which will ensure the token expires at the Max TTL time. This prevents 
+Team tokens living past their max ttl if Vault is unable to revoke the token.
+Omitting a max_ttl will Vault's system max_ttl
 
-Refer to [HCP Terraform Secrets
-Engine](/vault/tutorials/secrets-management/terraform-secrets-engine)
-for a step-by-step tutorial.
+In HCP Terraform, team tokens cannot have matching descriptions. In order to avoid
+collissions, we generate a random string as a suffix to the description. It is
```

Comment:
”We" usually means HashiCorp. In this case, "Vault" or perhaps, "this secrets engine" maybe?

```suggestion
collissions, Vault generates a random string as a suffix to the description. It is
```

Reply:
```suggestion 
collisions, the secret engine generates a random string as a suffix to the description. It is
```